### PR TITLE
Metalfoam structures can now be used as girders to finish building walls.

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
Let Metal foam structures work as a "girder" if you use metal sheets on them, finishing into a normal wall.

## Why It's Good For The Game
Rebuilding an area after an explosion without an RCD is slow, metal foam helps a lot if you don't have an RCD but you lose time on the clean up.
This cuts one step down at least, you can finish a wall on top of foam instead of breaking the foam barrier down and building a girder/wall from scratch in its place, possibly letting more of the atmos get spaced while doing it...

## Changelog

:cl: Guillaume Prata
qol: Metalfoam structures can now be used as a girder to finish building walls.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 1]
```diff
@@ -305,14 +307,70 @@
...
+	var/obj/item/stack/sheet/sheet_for_plating = W
+	if(istype(sheet_for_plating, /obj/item/stack/sheet/iron))
+		if(sheet_for_plating.get_amount() < 2)
+			to_chat(user, span_warning("You need two sheets of iron to finish a wall!"))
+			return
+		to_chat(user, span_notice("You start adding plating to the foam structure..."))
+		if (do_after(user, 40*platingmodifier, target = src))
+			if(sheet_for_plating.get_amount() < 2)
+				return
+			sheet_for_plating.use(2)
```
Comment (zxaber) [inline code/game/objects/effects/effect_system/effects_foam.dm:332]: IIRC, `use()` already checks if the stack has at least the specified amount. So you could just do

```
if(!sheet_for_plating.use(2))
  return

Comment (GuillaumePrata) [reply code/game/objects/effects/effect_system/effects_foam.dm:332]: Done and thanks.

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 2]
```diff
@@ -305,14 +307,70 @@
...
+	var/obj/item/stack/sheet/sheet_for_plating = W
+	if(istype(sheet_for_plating, /obj/item/stack/sheet/iron))
+		if(sheet_for_plating.get_amount() < 2)
+			to_chat(user, span_warning("You need two sheets of iron to finish a wall!"))
+			return
+		to_chat(user, span_notice("You start adding plating to the foam structure..."))
+		if (do_after(user, 40*platingmodifier, target = src))
+			if(!sheet_for_plating.use(2))
+				return
+			sheet_for_plating.use(2)
```
Comment (Ryll-Ryll) [inline code/game/objects/effects/effect_system/effects_foam.dm:332]: This second use() is redundant. kill it and we're good

**Comment by Fikou** [comment]
if i break the wall, will it leave the girder behind?
if not, maybe change the girder_type on the newly formed wall to the foam type?

**Comment by GuillaumePrata** [comment]
> if i break the wall, will it leave a girder behind?
if not, maybe change the girder_type on the newly formed wall to the foam type?

It works the same way it does for foam plating, it becomes the base item.
If you deconstruct this wall it will have a normal girder under it.

**Pull request merged**